node_modules/oae-principals/lib/test/util.js

```diff
@@ -1004,28 +1008,104 @@ var getAllGroupMembers = module.exports.getAllGroupMembers = function(restCtx, g
  * @param  {Function}       callback            Invoked when the user has been updated
  * @param  {User}           callback.user       The updated user profile
+ * @param  {String}         callback.token      The email verification token
  * @throws {AssertionError}                     Thrown if the update is unsuccessful or the expected results of updating a user don't hold true
  */
 var assertUpdateUserSucceeds = module.exports.assertUpdateUserSucceeds = function(restCtx, userId, update, callback) {
-    RestAPI.User.updateUser(restCtx, userId, update, function(err) {
+    // Get the user's profile to determine if there is an update to the email address, as we might
+    // have to wait until the verification email is sent out
+    RestAPI.User.getUser(restCtx, userId, function(err, user) {
         assert.ok(!err);
+        var isEmailChange = (update.email && update.email.toLowerCase() !== user.email);
 
-        // Assert the changes were persisted correctly
-        RestAPI.User.getUser(restCtx, userId, function(err, user) {
+        // Determine if the current user is an administrator
+        RestAPI.User.getMe(restCtx, function(err, me) {
             assert.ok(!err);
-            _.each(update, function(value, key) {
-                assert.strictEqual(user[key], value);
+            var isAdmin = (me.isTenantAdmin || me.isGlobalAdmin);
+
+            // If an administrator performs the update, there won't be a verification email
+            var expectVerificationEmail = (isEmailChange && !isAdmin);
+
+            var user = null;
+            var token = null;
+            var done = _.after(2, function() {
+                return callback(user, token);
+            });
+
+            // Update the user profile
+            RestAPI.User.updateUser(restCtx, userId, update, function(err) {
+                assert.ok(!err);
+
+                // Assert the changes were persisted correctly
+                RestAPI.User.getUser(restCtx, userId, function(err, _user) {
+                    assert.ok(!err);
+                    _.each(update, function(value, key) {
+                        // Email changes only get reflected in the user profile if they are done by administrators
+                        if (key !== 'email') {
+                            assert.strictEqual(_user[key], value);
+                        }
+                    });
+
+                    user = _user;
+                    done();
+                });
             });
 
-            return callback(user);
+            // Only attach a listener when we expect a verification email to be sent out
+            OaeUtil.invokeIfNecessary(expectVerificationEmail, onceVerificationEmailSent, update.email, function(_token) {
+                token = _token;
+                done();
+            });
```

Comment:
I think we probably want to verify tenant admin and global admin email addresses too. They can also make mistakes when typing.